### PR TITLE
added CEUS downloader

### PR DESCRIPTION
**Pull request by berkserbet**
### Why:
* This is the downloader for CEUS files.

### What:
* This PR adds the functions to download CEUS files given FCZ names, Buildingtype names and download url. 

### Checklist
- [ ] Finished My Changes
- [ ] Automated Unit Tests

### CODE REVIEW

File: CEUS/CEUS_download.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
```
Comment (jongoncalves) [inline CEUS/CEUS_download.py]: Add more appropriate header information, e.g.:
```
__author__ = "Rob Knight"
__copyright__ = "Copyright 2019, SLAC"
__license__ = "GPL"
__version__ = "1.0.1"
__maintainer__ = "Rob Knight"
__email__ = ""
__status__ = "Beta | Alpha"
```

Not necessarily all of these, but you get the point.

File: CEUS/CEUS_download.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This module downloads Commercial End-Use Survey (CEUS) files into intended filepath.
+"""
+
+import requests
+
+def CEUS_download(FCZs, buildingtypes, ceus_url, filepath):
```
Comment (jongoncalves) [inline CEUS/CEUS_download.py]: Is there anything unique about CEUS download v.s. RBSA downloads - or any future downloads for that matter?

We can make this generic where a list of URLs get passed in and the logic that is specific to CEUS remains the onus of the caller of the download function. A download file class doesn't need to know about CEUS filename construction.

File: CEUS/CEUS_download.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        for buildingtype in buildingtypes:
+
+            filename = FCZ+'_'+buildingtype+'.xls'
+            download_url = ceus_url+filename
+
+            if is_downloadable(download_url):
+                file = requests.get(download_url)
+                open(filename, 'wb').write(file.content)
+                
+    return
```
Comment (jongoncalves) [inline CEUS/CEUS_download.py]: This empty return is unnecessary

File: CEUS/CEUS_download.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    if 'text' in content_type.lower():
+        raise Exception("CEUS file is not downloadable from following link:", url)
+        return False
+
+    if 'html' in content_type.lower():
+        raise Exception("CEUS file is not downloadable from following link:", url)
+        return False
+
+    return True
```
Comment (jongoncalves) [inline CEUS/CEUS_download.py]: I like this function. No need to raise an exception however - which in this case prevents the return statement from being reached during code execution.

Add a logging routine (I guess print statement is ok in the short term) and then return `False`

**Comment by jongoncalves** [comment]
Also, a generic download file class/def/function should be removed from the CEUS folder. Put it in the root for now.

One last thing, add the `requests` lib dependency to the `loadinsight-environment.yml`

**Comment by jongoncalves** [comment]
> It seems like you're starting from the bottom up without a clear definition of the structure of what we're doing. Let's talk the time to do some architecting first based on this as a use case, e.g.,
> 
> ```
>   ceus_source_specification = LoadInsightStorageSpecification(url="http://someplace/")
>   input = LoadInsightDataObject(ceus_source_specification)
>   output = download_ceus(input)
>   output.write()
> ```
> 
> That means we need to design the LoadInsightStorageSpecification class and LoadInsightDataObject class, and come up with a function template that every function will use.

I'll put the base of the work for this together...

**Comment by dchassin** [comment]
Take a look at the draft PR#2. I’m stubbing out DataObject, Transformation, and Pipeline.  What I don’t know is how we will deal with remote storage.  I think we need a class that maintains the sync between the local copies and the remote “official” copies.  Kind of like a cache manager.

D

On Jun 5, 2019, at 12:32 PM, Jonathan Goncalves <notifications@github.com<mailto:notifications@github.com>> wrote:


It seems like you're starting from the bottom up without a clear definition of the structure of what we're doing. Let's talk the time to do some architecting first based on this as a use case, e.g.,

  ceus_source_specification = LoadInsightStorageSpecification(url="http://someplace/")
  input = LoadInsightDataObject(ceus_source_specification)
  output = download_ceus(input)
  output.write()


That means we need to design the LoadInsightStorageSpecification class and LoadInsightDataObject class, and come up with a function template that every function will use.

I'll put the base of the work for this together...

**Comment by jongoncalves** [comment]
Closing this for now.

**Pull request closed without merging**